### ISSUE TITLE
react-hot-loader with React.lazy not working

### ISSUE THREAD

**Issue by Blanceluo** [gaearon/react-hot-loader#1447]
### Description
react-hot-loader with React.lazy not working
Lazy loading component changes page cannot respond to changes

### Expected behavior
When the lazy loading component changes, the page should respond to the change

### Actual behavior
You need to refresh the page manually to see the new changes

### Environment

React Hot Loader version: 4.12.20

Run these commands in the project folder and fill in their results:

1.  `node -v`: v12.16.1
2.  `npm -v`: 6.13.4

Then, specify:

1.  Operating system: win10 x64
2.  Browser and version: Google chrome 81

### Reproducible Demo
index.js
```
import React from 'react'
import ReactDom from 'react-dom'

import App from './views/App'

ReactDom.render(<App />, document.getElementById('root'))
```

app.js
```
import React, { Suspense, lazy } from 'react'

import { hot } from 'react-hot-loader/root'
import { HashRouter as Router, Route, Switch, Link } from 'react-router-dom'

const Home = lazy(() => import('@/views/Home/index'))
const Index = lazy(() => import('@/views/Index/index'))

class App extends React.Component {
  render() {
    return (
      <div>
        <div className="text">ceshi</div>
        <Router>
          <Link to="/index">index</Link>
          <Suspense fallback={<div>Loading</div>}>
            <Switch>
              <Route path="/" exact component={Home}></Route>
              <Route path="/index" exact component={Index}></Route>
              <Route component={Index}></Route>
            </Switch>
          </Suspense>

        </Router>
      </div>
    )
  }
}


export default hot(App)
```
webpack.config.js
```
devServer: {
    contentBase: path.resolve(__dirname, '../dist'),
    hot: true,
    host: '127.0.0.1',
    port: 8000,
    open: true,
    quiet: true,
    compress: true
  },

rules: [
 {
     test: /\.m?jsx?$/,
     exclude: /(node_modules|bower_components)/,
     use: [
            {
              loader: 'babel-loader'
            }
        ]
    }
]

plugins: [
    new FriendlyErrorsWebpackPlugin({
      compilationSuccessInfo: {
        messages: ['You applicaton is running here http://127.0.0.1:8000']
      }
    }),
    new webpack.NamedModulesPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
    new webpack.HotModuleReplacementPlugin()
  ]
```

**Comment by theKashey** [comment]
Yep, there are some reported problems around it.

**Comment by Blanceluo** [comment]
> 是的，周围有一些报告的问题：
> 
> * [＃1425](https://github.com/gaearon/react-hot-loader/issues/1425)

you can use my project and reappear it : https://github.com/Blanceluo/webpacks
thanks you!

> Yep, there are some reported problems around it:
> 
> * #1425

**Comment by Blanceluo** [comment]
> Yep, there are some reported problems around it:
> 
> * #1425

According to *#1425 it has been solved.

This plugin is not introduced, babel-plugin-dynamic-import-node

**Comment by xyy94813** [comment]
@Blanceluo 
How did you solve the problem????
Can you show an example?
I run the #1425 example with error in browser.

My core stack:

- CRA, ps: react-scripts v3.0.1.
- reate-app-rewired
- react-app-rewire-hot-loader

I add `babel-plugin-dynamic-import-node` and set `trackTailUpdates` to `false` too.
But hot loader is still not work with code splitting. 

react-scripts use `babel-preset-react-app`.
And it use '@babel/plugin-syntax-dynamic-import' in dev env.
and use `babel-plugin-dynamic-import-node` in test env.

Should i rm the babel plugin '@babel/plugin-syntax-dynamic-import' ?

**Comment by Blanceluo** [comment]
> @Blanceluo
> 您是如何解决问题的？？？？
> 你能举个例子吗？
> 我运行[＃1425](https://github.com/gaearon/react-hot-loader/issues/1425)示例，但浏览器中出现错误。
> 
> 我的核心堆栈：
> 
> * CRA，ps：react-scripts v3.0.1。
> * reate-app-wired
> * react-app-rewire-hot-loader
> 
> 我添加`babel-plugin-dynamic-import-node`并设置`trackTailUpdates`为`false`。
> 并使用`@hot-loader/react-dom`
> 
> ```js
> const  _rewireReactHotLoader  =（config，env）=> {
>    如果（env ===  ' development '）{
>      config。解决。别名 [ ' react-dom ' ] =  ' @ hot-loader / react-dom ' ;
>     // babel-preset-react-app在开发人员中使用它
>     //热加载器需要babel-plugin-dynamic-import-node 
>     removeInternalBabelPlugin（' @ babel / plugin-syntax-dynamic-import '）（
>       配置
>       环保
>     ）;
>     addBabelPlugin（
>        ' dynamic-import-node '  //热加载器需要它
>     ）（配置，环境）;
>   }
> 
>   返回 rewireReactHotLoader（config，env）;
> };
> ```
> 
> 但是热加载器仍然不能与代码拆分一起使用。
> 
> react-scripts使用`babel-preset-react-app`。
> 它在开发环境中使用了“ @ babel / plugin-syntax-dynamic-import”。
> 并用于`babel-plugin-dynamic-import-node`测试环境
> 
> 我应该使用babel插件'@ babel / plugin-syntax-dynamic-import'吗？

You may also need to introduce react-hot-loader， @hot-loader/react-dom just to support hooks...

**Comment by xyy94813** [comment]
Someone say it maybe `@hot-loader/react-dom` problem.
https://github.com/hot-loader/react-dom/issues/8#issuecomment-620611639

But, It is not work for me without using `@hot-loader/react-dom`.

I found a interesting thins when i debugger:

1. Change a lazy load component. 
2. HotLoaderClient will receive a WS msg success.
3. And `react-hot-loader` will target update function; The root component forceUpdate()
4. log warn: `React-Hot-Loader: some components were updated out-of-bound. Updating your app to reconcile the changes.`
5. Nothing change in page.

https://github.com/gaearon/react-hot-loader/blob/5e226f40aea2f995adbb446199a9cb93656ef465/src/hot.dev.js#L54-L82

If i wrap lazy load component by `hot`,  `module.instances` include the lazy load component instance. then everything work well.

There are some question:

- Should i wrap any component if it will lazy load?
- And, why not re-render lazy load component after root component forceUpdate.
